### PR TITLE
NF: Cleanup of SeekBarPreferenceCompat

### PR DESCRIPTION
**Pull request by BrayanDSO**
I actually didn't want to clean this, but got in the mid of it in order to create some tests

## Approach
One cleanup per commit

## How Has This Been Tested?

Ran the app and tweaked some seekbar preferences

## Checklist
_Please, go through these checks before submitting the PR._

- [X] You have a descriptive commit message with a short title (first line, max 50 chars).
- [X] You have commented your code, particularly in hard-to-understand areas
- [X] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/preferences/SeekBarPreferenceCompat.kt [hunk 1]
```diff
@@ -227,8 +216,19 @@ constructor(
                 layout.addView(mSeekLine, paramsSeekbar)
             }
             preference.setupTempValue()
-            mSeekBar.max = preference.relativeMax
-            mSeekBar.progress = preference.relativeProgress
+
+            mSeekBar = SeekBar(context).apply {
+                setOnSeekBarChangeListener(this@SeekBarDialogFragmentCompat)
+                max = preference.relativeMax
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/preferences/SeekBarPreferenceCompat.kt:202]: reviewer 2: `getPreference()` was a slightly complex call which was factored out to a variable.

I agree the reduction in code complexity is worth it in this case

File: AnkiDroid/src/main/java/com/ichi2/preferences/SeekBarPreferenceCompat.kt [hunk 2]
```diff
@@ -88,55 +87,38 @@ constructor(
...
-        }
-    }
-
     var value: Int
-        get() = if (mValue == 0) {
-            getPersistedInt(default)
-        } else {
-            mValue
-        }
+        get() = getPersistedInt(default)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/preferences/SeekBarPreferenceCompat.kt]: Original code is from 2011

:warning: I don't feel this was non-functional, we persist the value at different times (especially persisting the value earlier), but this seems simpler and correct

Previously we did this during `onProgressChanged`. with `fromUser = true`

Now we do this on all changes to `mValue`. We also no longer take `shouldPersist()` into account

**Review by david-allison — CHANGES_REQUESTED** [review]
Could you confirm you've tested one commit with functional changes.

Ideally a few unit tests around here or `@NeedsTest` to confirm new behaviour

I'd accept this PR with the mentioned commit dropped and split off into a separate PR

**Comment by BrayanDSO** [comment]
I think I'll drop the whole PR and start using androidx SeekBarPreference. We aren't on 2011 anymore when someone had to do their own implementation of a seek bar preference

**Comment by david-allison** [comment]
Strong approve. Didn't realise it was now Android native

**Comment by BrayanDSO** [comment]
I'm going to keep this one. I can get replace of most of the previous usages, as shown on the screenshot below, but not `Start of the next day` preference, so I'll keep the class while `Start of the next day` isn't migrated to a different class (a TimePreference with hour-only set would be the best I guess)

image

File: AnkiDroid/src/main/java/com/ichi2/preferences/SeekBarPreferenceCompat.kt (hunk 2 above)
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/preferences/SeekBarPreferenceCompat.kt]: Dropped the commit

**Comment by BrayanDSO** [comment]
Dropped the perhaps functional commits

**Comment by mikehardy** [comment]
For what it's worth, in my opinion the "start of next day" preference seems like it would be a fit for this idea of yours:

> TimePreference with hour-only set

**Review by mikehardy — APPROVED** [review]
LGTM - I don't think there's anything functional left and that was the only hangup prior

**Pull request merged**